### PR TITLE
Add PlatformSliverAppBar class with example

### PR DESCRIPTION
**Pull request by bookshiyi**
Inspired by PR #425.

### CODE REVIEW

File: lib/src/platform_sliver_app_bar.dart [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+      brightness: data?.brightness,
+      padding: data?.padding,
+      transitionBetweenRoutes: data?.transitionBetweenRoutes ?? true,
+      heroTag: data?.heroTag ?? _defaultHeroTag,
+    );
+  }
+}
+
+//! Copied from file: /opt/homebrew/Caskroom/flutter/3.10.0/flutter/packages/flutter/lib/src/cupertino/nav_bar.dart
+const Color _kDefaultNavBarBorderColor = Color(0x4D000000);
```
Comment (bookshiyi) [inline lib/src/platform_sliver_app_bar.dart:262]: Regarding these local variables, I didn't find a better solution, so I had to copy and paste them here intact.

File: lib/src/platform_sliver_app_bar.dart [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    return other is _HeroTag && other.navigator == navigator;
+  }
+
+  @override
+  int get hashCode => identityHashCode(navigator);
+}
```
Comment (bookshiyi) [inline lib/src/platform_sliver_app_bar.dart:297]: Regarding these local variables, I didn't find a better solution, so I had to copy and paste them here intact.

**Review by aqwert — APPROVED** [review]
Thanks

**Comment by aqwert** [comment]
Could you rebase. Sorry for the time it has taken to review

**Review by aqwert — CHANGES_REQUESTED** [review]
Can you rebase please and resolve the conflicts

**Comment by bookshiyi** [comment]
> Can you rebase please and resolve the conflicts

The conflicts has been resolved without rebase. How do you think about this?

**Review by aqwert — APPROVED** [review]
Thanks.LGTM

**Pull request merged**